pkg/proxy/service/cluster_service.go

```diff
@@ -63,40 +66,59 @@ func (s *ServiceImpl) GetNamespace(ctx *gin.Context, userID, username string) (*
 			targetNamespace := signup.CompliantUsername
 			saName := fmt.Sprintf("appstudio-%s", signup.CompliantUsername)
 			saNamespacedName := types.NamespacedName{Namespace: targetNamespace, Name: saName}
-			sa := &v1.ServiceAccount{}
-			if err := member.Client.Get(context.TODO(), saNamespacedName, sa); err != nil {
+			apiURL, err := url.Parse(member.APIEndpoint)
+			if err != nil {
 				return nil, err
 			}
-
-			for _, secret := range sa.Secrets {
-				secretNamespacedName := types.NamespacedName{Namespace: targetNamespace, Name: secret.Name}
-				s := &v1.Secret{}
-				log.Info(ctx, fmt.Sprintf("Getting secret %v", secretNamespacedName))
-				if err := member.Client.Get(context.TODO(), secretNamespacedName, s); err != nil {
-					return nil, err
-				}
-				if s.Annotations["kubernetes.io/created-by"] == "openshift.io/create-dockercfg-secrets" {
-					// There are two secrets/tokens for the SA and both are valid
-					// but let's always use the non-docker one for the sake of consistency
-					log.Info(nil, fmt.Sprintf("Skipping docker secret with the creted-by label: %v", secretNamespacedName))
-					continue
-				}
-				decodedToken, ok := s.Data["token"]
-				if !ok {
-					log.Info(ctx, fmt.Sprintf("Skipping secret with no data.token: %v", secretNamespacedName))
-					continue // It still might be the docker configuration token even if it doesn't have the "kubernetes.io/created-by" annotation
-				}
-				tokenStr := string(decodedToken)
-
-				apiURL, err := url.Parse(member.APIEndpoint)
-				if err != nil {
-					return nil, err
-				}
-				return namespace.NewNamespaceAccess(*apiURL, tokenStr, member.Client), nil
+			cl, err := newRESTClient(ctx, member)
+			if err != nil {
+				return nil, err
 			}
-			return nil, errs.New("no SA found for the user")
+			tokenStr, err := getServiceAccountToken(cl, saNamespacedName)
+			if err != nil {
+				return nil, err
+			}
+			return namespace.NewNamespaceAccess(*apiURL, tokenStr, member.Client), nil
 		}
 	}
 
 	return nil, errs.New("no member cluster found for the user")
 }
+
+func newRESTClient(ctx *gin.Context, c *cluster.CachedToolchainCluster) (*rest.RESTClient, error) {
+	startTime := time.Now()
+	defer func() {
+		log.Info(ctx, fmt.Sprintf("duration to init RESTClient: %s", time.Since(startTime).String()))
```

Comment:
@alexeykazakov in the e2e tests for the proxy, the log statement above produces the following output:
```
{"level":"info","ts":"2022-09-06T07:58:51.837Z","logger":"registration-service","msg":"duration to init RESTClient: 6.445µs","timestamp":"Tue, 06 Sep 2022 07:58:51 +0000","commit":"f170083","user_id":"c5594394-8deb-4968-baac-e8c83fee229a","username":"proxymember1"}
...
{"level":"info","ts":"2022-09-06T07:59:17.117Z","logger":"registration-service","msg":"duration to init RESTClient: 10.455µs","timestamp":"Tue, 06 Sep 2022 07:59:17 +0000","commit":"f170083","user_id":"02cff24e-7891-4fa3-bf46-e1e011d5f10e","username":"proxymember2"}
```
so I don't think we need to cache the client for now.